proto/generate.proto

```diff
@@ -70,6 +70,10 @@ message NextTokenChooserParameters {
     float frequency_penalty = 9;
     /// token watermarking using "A Watermark for Large Language Models"
     bool watermark = 8;
+    /// grammar (applied if not empty)
+    string grammar = 10;
+    /// fsm_grammar_state
+    uint32 fsm_grammar_state = 11;
```

Comment:
I'm not sure we should pass the grammar_state here. It's always 0 when it's starting out, right ?

Adding anything here adds more work on other backends, so we should keep it to a strict minimum.